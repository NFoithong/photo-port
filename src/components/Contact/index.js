import React, { useState } from 'react';

function ContactForm() {
// jsx
    const [formState, setFormState] = useState({
        name:"",
        emial:"",
        message:""
    });

    const { name, email, message } = formState;

    function handleChange(e){
        setFormState({...formState, [e.target.name]: e.target.value});
    }

    function handleSubmit(e) {
        e.preventDefault();
    }

    return (
        <section>
        <h1>Contact me</h1>
            <form onSubmit={handleSubmit} id="Contact-form">
            {/* name */}
                <div>
                    <label htmlFor="name">Name:</label>
                    <input onChange={handleChange} type="text" name="name" defaultValue={name}/>
                </div>
            {/* email */}
                <div>
                    <label htmlFor="email">Email address:</label>
                    <input onChange={handleChange} type="text" name="email" defaultValue={email}/>
                </div>
            {/* message */}
                <div>
                    <label htmlFor="message">Message:</label>
                    <textarea onChange={handleChange} name="message" row="5" defaultValue={message} />
                </div>
                <button type="submit">Submit</button>
            </form>
        </section>
    )
}

export default ContactForm;