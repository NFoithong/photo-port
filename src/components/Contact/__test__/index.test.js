import React from "react";
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Contact from "..";

afterEach(cleanup);

describe('Contact component renders', ()=>{
    it('renders', ()=>{
        render(<Contact />)
    });

    it('render', ()=>{
        const { asFragment } = render(<Contact />)
        expect(asFragment).toMatchSnapshot()
    })
})

it('renders', ()=>{
    const { getByTextId } = render(<Contact />)
    expect(getByTextId('h1tag')).toHaveTextContent('Contact me')
})

it('render', ()=>{
    const { getByTextId } = render(<Contact />)
    expect(getByTextId('button')).toHaveTextContent('Submit')
})