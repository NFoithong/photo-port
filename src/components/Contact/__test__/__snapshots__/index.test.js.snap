// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component renders render 1`] = `[Function]`;
